src/components/Link/Link.tsx

```diff
@@ -0,0 +1,69 @@
+import classnames from 'classnames';
+
+import { px } from '../../utils';
+import React, { HTMLAttributes } from 'react';
+import { getLinkVariantClassName } from './utils';
+
+export const LinkVariants = {
+  /** Default variant, used */
+  standalone: 'standalone',
+  /** link rendering emailto: */
+  email: 'email',
+  /** these links are being rendered in a list */
+  list: 'list',
+  /** link is being rendered within body copy */
+  inline: 'inline',
+} as const;
+
+export interface LinkProps extends HTMLAttributes<HTMLAnchorElement> {
+  /**
+   * Describes where the link is used. It controls the styling of the link so we apply consistent styles. Defaults to `standalone`.  See the documentation [here](https://www.figma.com/file/xMuOXOAKVt5HC7hgYjF3ot/Components-v2.0?type=design&node-id=5731-12815) to see where each variant is used.
+   *
+   * @default standalone
+   * @see LinkVariants
+   */
+  variant?: keyof typeof LinkVariants;
+  /**
+   * The text of the link
+   */
+  children: React.ReactNode;
+  /**
+   * URL to navigate to when clicked
+   */
+  href: string;
+  /**
+   * Can be used to render alternative link components like the prefetching `Link` from `@remix-run/react`.
+   * If you implement this you should handle the `children`, `dataTestId`, `id`, `className`, and `href` props.
+   */
+  component?: React.ElementType<LinkProps & { 'data-testid': string }>;
```

Comment:
This was going to be `as` or `element`.  I am also not getting any TS errors for `data-testid` when omitting this in the one I did for Grid
```js
element?: React.ElementType<GridProps>;
```